### PR TITLE
MAT-4643: Changed security, for grant Access to a Measure, so that it…

### PR DESCRIPTION
**Pull request by gregory-akins**
… uses and API key like the MAT Measure Transfer endpoint)

## MADiE PR

Jira Ticket: [MAT-4643](https://jira.cms.gov/browse/MAT-4643)
(Optional) Related Tickets:

### Summary
Changed the endpoint so it uses an API key instead of a User Token
### All Submissions
* [X] This PR has the JIRA linked.
* [X] Required tests are included.
* [X] No extemporaneous files are included (i.e Complied files or testing results).
* [X] This PR is merging into the **correct branch**.
* [X] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/cms/gov/madie/measure/resources/MeasureController.java [hunk 1]
```diff
@@ -154,19 +157,22 @@ public ResponseEntity<String> updateMeasure(
...
       @PathVariable("id") String id,
       @RequestParam(required = true, name = "userid") String userid,
+      @Value("${lambda-api-key}") String apiKey,
       Principal principal) {
     ResponseEntity<String> response = ResponseEntity.badRequest().body("Measure does not exist.");
-    final String username = principal.getName();
-    log.info("getMeasureId [{}] by user [{}]", id, username);
 
-    if (measureService.grantAccess(id, userid, username)) {
+    log.info("getMeasureId [{}] using apiKey ", id, apiKey);
```
Comment (jkotanchik-SB) [inline src/main/java/cms/gov/madie/measure/resources/MeasureController.java]: Should we avoid logging the key?

File: src/main/java/cms/gov/madie/measure/resources/MeasureController.java [hunk 2]
```diff
@@ -154,19 +157,22 @@ public ResponseEntity<String> updateMeasure(
...
+
+    if (measureService.grantAccess(id, userid, apiKey)) {
       response =
           ResponseEntity.ok()
               .body(String.format("%s granted access to Measure successfully.", userid));
-      actionLogService.logAction(id, Measure.class, ActionType.UPDATED, username);
+      actionLogService.logAction(id, Measure.class, ActionType.UPDATED, apiKey);
```
Comment (jkotanchik-SB) [inline src/main/java/cms/gov/madie/measure/resources/MeasureController.java]: Same question on whether we should log the key.

File: src/test/java/cms/gov/madie/measure/resources/MeasureControllerMvcTest.java [hunk 3]
```diff
@@ -64,6 +77,12 @@ public class MeasureControllerMvcTest {
   @Captor private ArgumentCaptor<Measure> measureArgumentCaptor;
 
   private static final String TEST_USER_ID = "test-okta-user-id-123";
+
+  private static final String HARP_ID_HEADER_KEY = "harp-id";
+  private static final String HARP_ID_HEADER_VALUE = "XxYyZz";
+  private static final String LAMBDA_TEST_API_KEY_HEADER = "api-key";
+  private static final String LAMBDA_TEST_API_KEY_HEADER_VALUE = "9202c9fa";
```
Comment (jkotanchik-SB) [inline src/test/java/cms/gov/madie/measure/resources/MeasureControllerMvcTest.java]: Is there isn't a lambda associated with this controller, recommend removing the `LAMBDA` prefix.

**Review by jkotanchik-SB — DISMISSED** [review]

File: src/main/java/cms/gov/madie/measure/resources/MeasureController.java (hunk 1 above)
Comment (gregory-akins) [reply src/main/java/cms/gov/madie/measure/resources/MeasureController.java]: Yep.. good catch

**Review by jkotanchik-SB — APPROVED** [review]

**Pull request merged**